antenna/breakpad_resource.py

```diff
@@ -288,51 +288,55 @@ def get_throttle_result(self, raw_crash):
+
+            elif 'legacy_processing' in raw_crash and 'throttle_rate' in raw_crash:
+                # If we have throttle results for this crash and the values are
+                # valid, then use those instead of the throttle result
+                try:
+                    temp_result = int(raw_crash['legacy_processing'])
+                    if temp_result not in (ACCEPT, DEFER):
+                        raise ValueError('Result is not a valid value: %r', result)
+
+                    temp_throttle_rate = int(raw_crash['throttle_rate'])
+                    if not (0 <= temp_throttle_rate <= 100):
+                        raise ValueError('Throttle rate is not a valid value: %r', result)
+
+                    # The two values are valid, so this becomes our new
+                    # throttle result
+                    result = temp_result
+                    rule_name = 'ALREADY_THROTTLED'
+                    throttle_rate = temp_throttle_rate
+
+                except ValueError:
+                    # If we've gotten a ValueError, it means one or both of the
+                    # values is bad and we should ignore it and move forward
+                    mymetrics.incr('throttle.bad_throttle_values')
+
+            elif raw_crash.get('Throttleable', None) == '0':
+                # If the raw crash has ``Throttleable=0``, then it was
+                # submitted manually, so accept the crash instead of the
+                # throttle result
+                result = ACCEPT
+                rule_name = 'THROTTLEABLE_0'
                 throttle_rate = 100
 
-                # Save the results in the raw_crash itself
-                raw_crash['legacy_processing'] = result
-                raw_crash['throttle_rate'] = throttle_rate
-
-                return result, 'FROM_CRASHID', throttle_rate
-
-        # If we have throttle results for this crash, return those.
-        if 'legacy_processing' in raw_crash and 'throttle_rate' in raw_crash:
-            try:
-                result = int(raw_crash['legacy_processing'])
-                if result not in (ACCEPT, DEFER):
-                    raise ValueError('Result is not a valid value: %r', result)
-
-                throttle_rate = int(raw_crash['throttle_rate'])
-                if not (0 <= throttle_rate <= 100):
-                    raise ValueError('Throttle rate is not a valid value: %r', result)
-                return result, 'ALREADY_THROTTLED', throttle_rate
-
-            except ValueError:
-                # If we've gotten a ValueError, it means one or both of the
-                # values is bad and we should ignore it and move forward.
-                mymetrics.incr('throttle.bad_throttle_values')
-
-        # If we have a Throttleable=0, then return that.
-        if raw_crash.get('Throttleable', None) == '0':
-            # If the raw crash has ``Throttleable=0``, then we accept the
-            # crash.
-            mymetrics.incr('throttleable_0')
```

Comment:
I don't know why this did a separate metrics ping. It gets done again later as a tag which we have in our dashboards.